src/game/Unit.cpp

```diff
@@ -1452,25 +1456,49 @@ void Unit::CalculateMeleeDamage(Unit* pVictim, CalcDamageInfo* damageInfo, Weapo
             break;
     }
 
+    bool immune = true;
+
+    for (uint8 i = 0; i < m_weaponDamageCount[damageInfo->attackType]; i++)
+    {
+        SubDamageInfo *subDamage = &damageInfo->subDamage[i];
+
+        subDamage->damageSchoolMask = SpellSchoolMask(1 << GetWeaponDamageSchool(damageInfo->attackType, i));
```

Comment:
This conversion, I'm no longer sure about. The code in Classic is: 
```c++
subDamage->damageSchoolMask = GetSchoolMask(GetWeaponDamageSchool(damageInfo->attackType, i));
```
But `GetScoolMask()` only exists in Classic:
```c++
// it convert school value into schoolmask missing in 1.12 dbc
inline SpellSchoolMask GetSchoolMask(uint32 school)
{
    return SpellSchoolMask(1 << school);
}
```